File: Library_API/BusinessLogicLayer/DTOs/EBookCreateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DataAccessLayer.Enums;

namespace BusinessLogicLayer.DTOs
{
    public record struct EBookCreateRequest(
        string Title,
        string ISBN,
        string Publisher,
        int Year,
        int PagesCount,
        string Description,
        string CoverImageUrl,
        string FileUrl,
        BookAccessType BookAccessType,
        List<string> AuthorNames,
        List<string> GenreNames);
}

File: Library_API/UI/library-app/src/app/interfaces/ebook-create-request.ts
export interface EbookRequest {
    title: string,
    isbn: string,
    publisher: string,
    year: number,
    pagesCount: number,
    description: string,
    coverImageUrl: string,
    fileUrl: string,
    bookAccessType: string,
    authorNames: string[],
    genreNames: string[],
}
